src/firefly/js/ui/dynamic/EmbeddedPositionSearchPanel.jsx

```diff
@@ -412,6 +412,7 @@ function SpatialSearch({rootSlotProps: slotProps, insetSpacial, uploadInfo, setU
                 tooltip: 'Chose type of search', initialState: {value: initToggle},
                 options: searchTypeToggleOptions
             }} />}
+            {slotProps.searchTypeOptions}
```

Comment:
-  This is probably the wrong name. `searchTypeOptions` is confusing with `searchTypeOp` right below.  From `EmbeddedPositionSearchPanel`'s perspective it is just a widget that goes in `SpatialSearch`. I think a better name is `midDetailsWidget`
- This should be part of the `SpatialSearch` slotProps
- so `spatialSearch.MidDetailsWidget`
   - change line 405 to be
 ```js
      const { searchTypeKey=CONE_AREA_KEY, midDetailsWidget, sx } = slotProps.spatialSearch ?? {};
```
- This should be added to `EmbeddedPositionSearchPanel.propTypes`
- I did not put this in the UI comments but the spacing it too tight when you add and extra component. I think you change change the `spacing` with the slotProp: `spacialSearch.sx`

Reply:
I second what @robyww said. Just a suggestion: `midDetailsWidget` can be defined as `children` because it's basically a slot in this component to render another component.